### ISSUE TITLE
[FEAT] Auto-create Webhook on Github for mirror repositories

### ISSUE THREAD

**Issue by MB175** [go-gitea/gitea#22240]
### Feature Description

Automatically create a github webhook which calls the gitea api to trigger a sync job, the setup could look like that.

The GitHub fine granular Personal Access Token does allow to for example restrict access to create webhooks, which is ideal for this setup.

Setup migration as mirror in gitea as usually
gitea uses the github pat to check if the rights for editing webhooks is given
gitea makes web call to github and creates webhook with the following payload (gitea needs to create an api access token at this point) [GitHub REST API](https://docs.github.com/en/rest/webhooks?apiVersion=2022-11-28#repository-webhooks)
Call-Back-Url in direction of gitea

```
<host>/api/v1/repos/{owner}/{repo}/mirror-sync?access_token=< gitea_acces token>
```

The given setup depending on the settings of the webhook on github will always trigger a sync of the mirror when a push is done on github.

This would be a huge time saver if these steps would be done by the mirror wizard just by ticking a checkbox at creation.

**Comment by jolheiser** [comment]
But the mirrors already sync on a cron. What is the benefit of this?

**Comment by delvh** [comment]
I guess to sync once there are changes on GitHub?

**Comment by jolheiser** [comment]
> The given setup depending on the settings of the webhook on github will always trigger a sync of the mirror when a push is done on github.

My apologies, I overlooked this part.

**Comment by MB175** [comment]
@delvh 

> I guess to sync once there are changes on GitHub?

exactly, I alway set it up like that

**Comment by MB175** [comment]
@jolheiser 

> > The given setup depending on the settings of the webhook on github will always trigger a sync of the mirror when a push is done on github.
> 
> My apologies, I overlooked this part.
> 
> In my opinion, mirrors rarely need this level of timeliness, and something about setting up a webhook on github rubs me the wrong way, rather something that could be done by another system or CI.
> 
> Just my opinion on the matter.

Yes, you are right about that, I did it in a CI before, but the bloat is way bigger than just using the webhook.

In my opinion, it's very clean to use the web hook to trigger the sync, especially is helpful when you want to use a CI that's coupled to your Gitea, but the main project is hosted on GitHub.

I mean the functionality is already there, just the setup would be automized and simplified
